backend/src/users/users.controller.ts

```diff
@@ -16,14 +16,23 @@ export class UsersController {
         await sendEmail(user.email, url)
     }
 
+    @Get()
+    async findOne(@Query() query){
+        if(!query.password && !query.email) {
+            return await this.findAll()
```

Comment:
I use 3 get methods. One with the findAll, one with the query, and one with the id. When I am trying to access findAll or query one, the one that's above gets called.

Reply:
I think it would be the best to throw an error in this case. The name if the method is `findOne`, which means it should only find a single item or none.

Reply:
I'd change it to:
```
if (!query.password || !query.email) {
  throw new Error('Missing required fields')
}
```

Reply:
So, should I remove findAll method?

Reply:
Yes, method named `findOne` should not be returning every record.